lib/elixir/lib/calendar/datetime.ex

```diff
@@ -1674,6 +1674,67 @@ defmodule DateTime do
     end
   end
 
+  @doc """
+  Shifts a datetime by given duration according to its calendar.
+
+  Can return an ambiguous or gap datetime tuple.
```

Comment:
Let's add a bit more context here. We should also change the docs for DateTime.add to mention this function. :)